components/create_comment/create_comment.jsx

```diff
@@ -548,7 +559,29 @@ export default class CreateComment extends React.PureComponent {
         this.draftsForPost[this.props.rootId] = updatedDraft;
     }
 
+    handleMouseUp = (e) => {
+        e.persist();
+        const caretPosition = Utils.getCaretPosition(e.target);
+        const target = e;
+        this.setState({
+            caretPosition,
+            target,
+        });
+    }
+
     handleKeyDown = (e) => {
+        e.persist();
+
+        // A bit of a hack. Better to use keyup instead.
+        // We therefore use setTimeout to wait for the right value of caretPosition
+        // and then update state.
+
+        setTimeout(() => {
+            const caretPosition = Utils.getCaretPosition(e.target);
+            const target = e;
+            this.setState({caretPosition, target});
```

Comment:
Instead of storing the target in the state, what if we used a ref to access the inner textbox instead? The `Textbox` component already accesses the inner textbox in places [like this](https://github.com/mattermost/mattermost-webapp/blob/master/components/textbox/textbox.jsx#L125), so I don't know if it would be hard to expose that to the `CreateComment` and `CreatePost` components. State doesn't seem like the best place to store that reference since it's not used for rendering the component

Reply:
@hmhealey implemented the suggested changes on this and picked a `ref` from the inner textbox